### PR TITLE
[TT-4708] Refactor go plugin OAS conversion

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3951/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2022676783)

File: apidef/oas/security.go [hunk 1]
```diff
@@ -294,6 +294,22 @@ func (s *OAS) extractCustomPluginTo(api *apidef.APIDefinition) {
 	api.AuthConfigs[apidef.CoprocessType] = authConfig
 }
 
+func (s *OAS) fillGoPlugin(api apidef.APIDefinition) {
+	goPlugin := &GoPlugin{}
+	goPlugin.Enabled = api.UseGoPluginAuth
```
Comment (titpetric) [inline apidef/oas/security.go:299]: Could we just use this value in order to avoid the allocation if ShouldOmit(goPlugin)?

**Review by titpetric — APPROVED** [review]

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3951/merge
Commit: 19912f68f74b6612a6f9a0b3b4b5394890e00270 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2028496182)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 9a6848391f0f48ff171aff64ecea20e38c61fc7d Refactor go plugin OAS conversion (#3951)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2028886012)

**Pull request merged**